element-hq/element-x-android
# Push provider switch

## Type of change


## Content


Add an advanced setting entry to switch between available push providers: Firebase, and available UnifiedPush distributors.
Also properly unregister existing pushers and update the codebase for everything to work as expected.

Please note that matrix.org user may have issue getting push from UnifiedPush since there is some rate limiting. This will be handled separately.

## Motivation and context



## Screenshots / GIFs



![PushProviderSelector2](https://github.com/element-hq/element-x-android/assets/3940906/6bc60c2c-b2f5-49e0-a70a-238e96188783)


## Tests



- Install the application
- Install the application ntfy from https://f-droid.org/fr/packages/io.heckel.ntfy/ for instance
- Use the application and ensure that push is received
- Go to settings / advanced settings and switch the push provider to `ntfy`.
- The push should still be received (with the limitation on matrix.org)
- You can also check on the notification troubleshoot screen that Push are received (even for matrix.org users)
- It's also possible to check the registered pushers by using Element Android on the same account: go to setting / advanced settings / Notification targets to see the list of active pushers.

## Tested devices

- OS version(s):

## Checklist
